jffs2/libjffs2.c

```diff
@@ -1081,8 +1080,30 @@ int libjffs2_mount(storage_t *strg, storage_fs_t *fs, const char *data, unsigned
 }
 
 
-/* TODO: implement umount for jffs2 */
 int libjffs2_umount(storage_fs_t *fs)
 {
-	return -ENOSYS;
+	jffs2_partition_t *part;
+
+	if ((fs == NULL) || (jffs2_common.fs == NULL)) {
+		return -EINVAL;
+	}
+	part = (jffs2_partition_t *)fs->info;
+
+	/* TODO: is it safe to unmount? */
+	/* Check for open files, mountpoints within the filesystem etc. */
+
+	/* Destroy in-memory objects */
+	dev_done(part->devs);
+	object_done(part);
```

Comment:
Hmm, maybe we should want to do this after making sure that gc has been defeated?

Reply:
You're right. GC thread can access objects (e.g. via `iget()`\`iput()`) therefore `object_done()` should be called after GC thread was killed. Fixed.
Sidenote:
`put_super()` should also be called after we're done with the GC tread. I had to move it to `kill_mtd_super()`.